framework/src/consumption/lib/redshift-serverless/redshift-serverless-workgroup.ts

```diff
@@ -0,0 +1,278 @@
+// SPDX-License-Identifier: MIT-0
+import { createHash } from 'crypto';
+import { Names, RemovalPolicy, Stack } from 'aws-cdk-lib';
+import { Connections, IConnectable, IVpc, Port, SecurityGroup, SelectedSubnets } from 'aws-cdk-lib/aws-ec2';
+import { CfnConnection } from 'aws-cdk-lib/aws-glue';
+import { CfnWorkgroup } from 'aws-cdk-lib/aws-redshiftserverless';
+import { Construct } from 'constructs';
+import { RedshiftServerlessNamespace } from './redshift-serverless-namespace';
+import { RedshiftServerlessWorkgroupProps } from './redshift-serverless-workgroup-props';
+import { DataCatalogDatabase } from '../../../governance';
+import { Context, DataVpc, TrackedConstruct, TrackedConstructProps, Utils } from '../../../utils';
+import { RedshiftData } from '../redshift/redshift-data';
+
+const DEFAULT_PORT = 5439;
+
+/**
+ * Create a Redshift Serverless Workgroup. A default namespace would be created if none is provided.
+ *
+ * @example
+ * const workgroup = new dsf.consumption.RedshiftServerlessWorkgroup(this, "RedshiftWorkgroup", {
+ *    workgroupName: "redshift-workgroup"
+ * })
+ */
+export class RedshiftServerlessWorkgroup extends TrackedConstruct implements IConnectable {
+  private static readonly DEFAULT_VPC_CIDR = '10.0.0.0/16';
+  private static readonly DEFAULT_VPC_PRIVATE_SUBNET_NAME = 'Private';
+
+  /**
+   * The created workgroup
+   */
+  readonly cfnResource: CfnWorkgroup;
+
+  /**
+   * The name of the created workgroup
+   */
+  readonly workgroupName: string;
+
+  /**
+   * Connections used by Workgroup security group. Used this to enable access from clients connecting to the workgroup
+   */
+  readonly connections: Connections;
+
+  /**
+   * The associated namespace
+   */
+  readonly namespace: RedshiftServerlessNamespace;
+
+  /**
+   * The name of the associated namespace
+   */
+  readonly namespaceName: string;
+
+  /**
+   * The Glue connection associated with the workgroup. This can be used by Glue ETL Jobs to read/write data from/to Redshift workgroup
+   */
+  readonly glueConnection: CfnConnection;
+
+  private readonly primarySecurityGroup: SecurityGroup;
+  private readonly removalPolicy: RemovalPolicy;
+  private redshiftData?: RedshiftData;
+  private readonly vpc: IVpc;
+  private readonly selectedSubnets: SelectedSubnets;
+
+  constructor(scope: Construct, id: string, props: RedshiftServerlessWorkgroupProps) {
+    const trackedConstructProps: TrackedConstructProps = {
+      trackingTag: RedshiftServerlessWorkgroup.name,
+    };
+
+    super(scope, id, trackedConstructProps);
+    this.removalPolicy = Context.revertRemovalPolicy(scope, props.removalPolicy);
+    const initNamespaceDetails = this.initNamespace(props);
+    this.namespace = initNamespaceDetails.namespace;
+    this.namespaceName = initNamespaceDetails.namespaceName;
+
+    if (props.vpc && props.subnets) {
+      this.vpc = props.vpc;
+      this.selectedSubnets = props.vpc.selectSubnets(props.subnets);
```

Comment:
can we take the default subnet selection of the provided VPC if `props.subnets` is not configured?

Reply:
So if the `props.vpc` is provided, but not the `props.subnets` this one is tricky because we won't know how the vpc is created and what are the subnets inside. We can theoretically "guess" by using the [`SubnetType`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.SubnetType.html) but that's volatile. That's why I'm checking for both.